plugin/pkg/scheduler/factory/factory.go

```diff
@@ -182,13 +310,19 @@ func (c *ConfigFactory) deletePodFromCache(obj interface{}) {
 	switch t := obj.(type) {
 	case *api.Pod:
 		pod = t
+		if pod.Spec.NodeName != "" {
+			c.EquivalencePodCache.SendInvalidAlgorithmCacheReq(pod.Spec.NodeName)
+		}
 	case cache.DeletedFinalStateUnknown:
 		var ok bool
 		pod, ok = t.Obj.(*api.Pod)
 		if !ok {
 			glog.Errorf("cannot convert to *api.Pod: %v", t.Obj)
 			return
 		}
+		if pod.Spec.NodeName != "" {
```

Comment:
Instead of having this if everywhere, we should add a function to equivalenceCache, like: "invalidateCacheAfterPodChanges(*api.Pod)" (obviously with better name).